Orange/data/domain.py

```diff
@@ -283,25 +283,35 @@ def index(self, var):
         except KeyError:
             raise ValueError("'%s' is not in domain" % var)
 
-    def has_discrete_attributes(self, include_class=False):
+    def has_discrete_attributes(self, include_class=False, include_metas=False):
         """
-        Return `True` if domain has any discrete attributes. If `include_class`
-                is set, the check includes the class attribute(s).
+        Return `True` if domain has any discrete attributes. If
+        `include_class` is set, the check includes the class attribute(s). If
+        `include_metas` is set, the check includes the meta attributes.
         """
         if not include_class:
-            return any(var.is_discrete for var in self.attributes)
+            attributes = list(self.attributes) if not include_metas \
```

Comment:
a) Isn't the conversion to lists unnecessary?
b) This whole function can be shortened even further, e.g.:
```
vars = self.variables if include_class else self.attributes
vars += self.metas if include_metas else ()
return any(var.is_discrete for var in vars)
```